### PR TITLE
Fix point-cloud limiting for float logits

### LINKED ISSUE #28: Chamfer distance can OOM when there are less than 10k points
When fixing #24, #26 introduces a new bug. Specifically, `_limit` returns `Float[...]` (nominally logits) when callers may be expecting `Bool[...]`. This leads to OOM, as well as potential calculation errors if the output is interpreted as bool via `torch.where` and other checks.

### PR DESCRIPTION
**Pull request by pentene**
Fixes point-cloud limiting so float logits are converted to boolean occupancy masks before `as_points`. Logits are still used for weighted sampling, but `_limit` now consistently returns Bool tensors.

### CODE REVIEW

**Review by thetianshuhuang — CHANGES_REQUESTED** [review]
I think there's a high level change that's been missed. This implementation is implicitly changing the abstraction; instead of arbitrary passing to PointCloudMetric.as_points and being turned into points, we're assuming grid-like is the argument, and as_points is responsible for taking an occupancy grid and turning it into N-dimensional Cartesian points. So if/when we add Chamfer for a modality that breaks this assumption (e.g., depth images), the current design won't work. This is in part also an existing issue with the code.

As an idea, what if we move max_points into Chamfer2D and Chamfer3D, and put _limit as a @staticmethod? Then Chamfer2D, Chamfer3D can "own" the responsibility of both subsampling and generating points.

**Comment by thetianshuhuang** [comment]
Was looking into this a bit more. Maybe the right thing to do is to use a proper knn gather kernel as the backend. Most of the options seem kind of bad (pytorch3d is on maintenance mode, for example), but this looks decent: https://github.com/getkeops/keops

**Comment by pentene** [comment]
> Was looking into this a bit more. Maybe the right thing to do is to use a proper knn gather kernel as the backend. Most of the options seem kind of bad (pytorch3d is on maintenance mode, for example), but this looks decent: https://github.com/getkeops/keops

I think this could be implemented in a separate PR.

**Comment by pentene** [comment]
I adopted the idea of transferring the responsibility to `Chamfer2D` and `Chamfer3D`, but this also indicates that we will need to replicate this logic, or specifically the `max_points` and `_limit`, in future subclasses. I am re-requesting the review, but I think, as you've mentioned, it will be better to modify the backend so that we do not need `max_points` to guard againt OOM. Nevertheless, changing the backend will need much more tests and work, and I still believe it would be better if we create a separate PR to do that.

**Review by thetianshuhuang — APPROVED** [review]
LGTM
